sql/core/src/main/scala/org/apache/spark/sql/execution/command/DataWritingCommand.scala

```diff
@@ -96,4 +99,19 @@ object DataWritingCommand {
       sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY),
       metrics.values.toSeq)
   }
+
+  def assertEmptyRootPath(tablePath: URI, saveMode: SaveMode, hadoopConf: Configuration) {
+    if (saveMode != SaveMode.Overwrite && !SQLConf.get.allowNonEmptyLocationInCTAS) {
```

Comment:
Can we add a bit more comment to explain the save mode here? For SQL CTAS, the mode is always append?

Reply:
For SQL CTAS, the mode is always `ErrorIfExists`. 
In that case, I'll change the condition as  `if (saveMode == SaveMode.**ErrorIfExists** && !SQLConf.get.allowNonEmptyLocationInCTAS)